compiler/ballerina-parser/src/main/java/io/ballerina/compiler/syntax/tree/OnFailClauseNode.java

```diff
@@ -40,12 +41,12 @@ public Token failKeyword() {
         return childInBucket(1);
     }
 
-    public TypeDescriptorNode typeDescriptor() {
-        return childInBucket(2);
+    public Optional<TypeDescriptorNode> typeDescriptor() {
+        return optionalChildInBucket(2);
```

Comment:
are these changes updated manually? `OnFailClauseNodeModifier` in the same class has not been updated.
Don't manually update this.  Make `typeDescriptor` and `failErrorName` optional in `syntax_tree_descriptor.json` and run `.gradlew treegen`. 

when you update the `syntax_tree_descriptor.json`, update `api_gen_syntax_tree_descriptor.json` with the same changes.